roles/set_custom_fact/tasks/main.yml

```diff
@@ -145,3 +129,14 @@
 - name: Set variable generic_project_name
   set_fact: generic_project_name={{ ansible_hostname | regex_replace('-\d+', '') }}
   tags: [ 'master', 'custom', 'rename', 'update','idc_import','package_management','kalite_import' ]
+
+- name: Stat /dev/sda1
+  stat: path=/dev/sda1
+  register: sda1
+  tags: ['always']
+
+- name: Use sda1 as external hard drive if sda1 exists and on arm architecture
+  set_fact: ext_hdd={{ (ansible_architecture == "armhf" or ansible_architecture == "armv7l") and (sda1.stat.isblk | default (omit)) == True }}
+  when: firststart.stat.exists == false or firststart is undefined
```

Comment:
About set_fact 
> These variables will be available to subsequent plays during an ansible-playbook run, but will not be saved across executions even if you use a fact cache.

So it can work if you remove the `when:` condition.

Otherwise, you have to store the `ext_hdd` in a fact file as done here https://github.com/ideascube/ansiblecube/blob/oneUpdateFile/roles/set_custom_fact/tasks/main.yml#L18-L22

We have two options : 

1. We "always" test if an HDD is attached to the device and tasks will be executed depending the result of `ext_hdd`, this way it could be a little bit more "dynamic" and we could image to switch from sd storage to hdd storage. Data still have to be migrated somehow.
2. We test only one time at custom run if an HDD is attached to the device. We store the result as a fact in a fact file `/etc/ansible/facts.d/installed_software.fact` and we use it as it is. The device will have always the same state. 

I think option 1 will not work anyway because if an HDD exist it has to be registered in `/etc/fstab`, if once the device start without HDD it will lead to an issue as the attached HDD define in fstab won't be found. There is no use then of testing at each run this variable. Storing in fact file is the best option from my point of view @letompouce @swbsf what do you think ?